hetio/permute.py

```diff
@@ -48,8 +82,49 @@ def permute_graph(graph, multiplier=10, seed=0, metaedge_to_excluded=dict(), log
 
 def permute_pair_list(pair_list, directed=False, multiplier=10, excluded_pair_set=set(), seed=0, log=False):
     """
-    If n_perm is not specific, perform 10 times the number of edges of permutations
-    May not work for directed edges
+    Permute edges (of a single type) in a graph according to the XSwap function
+    described in https://doi.org/f3mn58. This method selects two edges and
+    attempts to swap their endpoints. If the swap would result in a valid edge,
+    the swap proceeds. Otherwise, the swap is skipped. The end result is that
+    node degree is preserved, but edges are shuffled, thereby losing their
+    original meaning.
+
+    Parameters
+    ----------
+    pair_list : list of tuples
+        List of edges to permute. Each edge is represented at a (source,
+        target) tuple. source and target represent nodes and can be any Python
+        objects that define __eq__. In other words, this function does not
+        assume any specific format for nodes. If the edges are from a bipartite
+        or directed graph, then all tuples must have the same alignment. For
+        example, if the edges represent the bipartite Compound-binds-Gene
+        relationship, all tuples should be of the form (compound, gene) and not
+        intermixed with (gene, compound) tuples. The only instance where order
+        of the source and target is not important is for an undirected edge
+        type where the source and target nodes are of the same type, such as
+        Gene-interacts-Gene.
+    directed : bool
+        Whether the edge should be considered directed. If False, a swap that
+        creates an a-b edge will be invalid if a b-a edge already exists.
+    multiplier : int or float
+        This is multiplied by the number of edges in pair_list to determine the
+        number of swaps to attempt.
```

Comment:
Does this mean that the function will maker `multiplier` attempts at swapping *each* edge?
Or simply that "number of edges" * `multiplier` is the total number of attempts at swapping that will be made?

Reply:
> Does this mean that the function will maker multiplier attempts at swapping each edge?

Not this.

> "number of edges" * multiplier is the total number of attempts at swapping that will be made?

Yes.